ambry-cloud/src/test/java/com.github.ambry.cloud/azure/AzureIntegrationTest.java

```diff
@@ -73,6 +82,31 @@
   private String propFileName = "azure-test.properties";
   private String tokenFileName = "replicaTokens";
 
+  /**
+   * Parameterized constructor.
+   * @param replicationCloudTokenFactory type of token factory used by {@link CloudDestination}
+   */
+  public AzureIntegrationTest(String replicationCloudTokenFactory) throws ReflectiveOperationException {
+    super();
+    this.replicationCloudTokenFactory = replicationCloudTokenFactory;
+    Properties properties = new Properties();
+    properties.setProperty("replication.cloud.token.factory", replicationCloudTokenFactory);
+    VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
+    ReplicationConfig replicationConfig = new ReplicationConfig(verifiableProperties);
+    findTokenFactory =
+        new FindTokenHelper(null, replicationConfig).getFindTokenFactoryFromReplicaType(ReplicaType.CLOUD_BACKED);
+  }
+
+  /**
+   * static method to generate parameters.
+   * @return {@link Collection} of parameters.
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> input() {
```

Comment:
Same comment as in unit test: only testFindEntriesSince() really needs this and other tests can be time consuming.

Reply:
I think it wouldn't be too bad to have parameterization in this class, as FindTokenHelper.getFindTokenFactoryFromReplicaType() also sort of gets tested here.